src/layer/riscv/rvv_mathfun.h

```diff
@@ -308,7 +308,7 @@ _RVV_FLOAT32_COS_OP(8, 4)
                                                                                                  \
         /* clamp the inputs to the range [-9, 9] since anything outside */                       \
         /* this range is -/+1.0f in single-precision.                   */                       \
-        x2 = vfmin_vf_f32m##LMUL(x, c_tanh_hi, vl);                                              \
+        x2 = vfmin_vf_f32m##LMUL(x2, c_tanh_hi, vl);                                             \
```

Comment:
and the same fix in rvv_mathfun_fp16s.h ?

Reply:
Yes, ditto.